### PR TITLE
test: Prevent SLT from falsely passing queries

### PR DESCRIPTION
**Pull request by jkosh44**
Previously, if a query was parsed successfully, and it was not a CREATE VIEW, SELECT, or SHOW query then it passed the SLT test no matter what.

This commit updates SLT to check the result of those queries.

### Checklist

- [ ] This PR has adequate test coverage / QA involvement has been duly considered.
- [ ] This PR has an associated up-to-date [design doc](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/design/README.md), is a design doc ([template](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/design/00000000_template.md)), or is sufficiently small to not require a design.
  
- [ ] If this PR evolves [an existing `$T ⇔ Proto$T` mapping](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/command-and-response-binary-encoding.md) (possibly in a backwards-incompatible way), then it is tagged with a `T-proto` label.
- [ ] If this PR will require changes to cloud orchestration, there is a companion cloud PR to account for those changes that is tagged with the release-blocker label ([example](https://github.com/MaterializeInc/cloud/pull/5021)).
  
- [ ] This PR includes the following [user-facing behavior changes](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/guide-changes.md#what-changes-require-a-release-note):
  -

### CODE REVIEW

File: src/sqllogictest/src/runner.rs [hunk 1]
```diff
@@ -1175,17 +1175,6 @@ impl RunnerInner {
             [statement] => statement,
             _ => bail!("Got multiple statements: {:?}", statements),
         };
-        match statement {
-            Statement::CreateView { .. }
-            | Statement::Select { .. }
-            | Statement::Show(ShowStatement::ShowObjects(..)) => (),
-            _ => {
-                if output.is_err() {
-                    // We're not interested in testing our hacky handling of INSERT etc
```
Comment (jkosh44) [inline src/sqllogictest/src/runner.rs:1184]: @benesch any idea what this comment means? My only guess is that it's from a time when `INSERT` wasn't implemented?

**Comment by jkosh44** [comment]
If anyone in a European timezone wants to take this over, then feel free. I haven't gone through all of the CI logs, but most of test failures look like tests that were previously being ignored and not their output is actually being checked.

**Review by def- — COMMENTED** [review]
I can fix the SLT failures that pop up from this.

File: src/sqllogictest/src/runner.rs (hunk 1 above)
Comment (mjibson) [reply src/sqllogictest/src/runner.rs:1184]: That's probably from 3 years ago when INSERT was very different. Is almost certainly wildly outdated and misleading.

Comment (benesch) [reply src/sqllogictest/src/runner.rs:1184]: Yep, Matt's right.

**Comment by benesch** [comment]
Thanks very much! Surprising how long this code has persisted.

**Review by def- — APPROVED** [review]

**Review by nrainer-materialize — APPROVED** [review]

File: src/sqllogictest/src/runner.rs (hunk 1 above)
Comment (philip-stoev) [reply src/sqllogictest/src/runner.rs:1184]: Yes, previously INSERT ... SELECT was not implemented in Mz. Instead, it was sent off to Postgres and the result re-ingested from there.

File: test/sqllogictest/list.slt [hunk 2]
```diff
@@ -1986,10 +1986,10 @@ SELECT '{1}'::float4 list = '{2}'::float8 list
 query error ELEMENT TYPE must be of class type, but received pg_catalog.pg_enum which is of class view
 CREATE TYPE tbl_list AS LIST (ELEMENT TYPE=pg_enum)
 
-query error CREATE TYPE ... AS LIST option "ELEMENT TYPE" cannot accept type modifier on numeric, you must use the default type
+query error CREATE TYPE ... AS LIST option ELEMENT TYPE cannot accept type modifier on pg_catalog.numeric, you must use the default type
 CREATE TYPE typ_mod_list AS LIST (ELEMENT TYPE=numeric(38,0))
 
-query error CREATE TYPE ... AS LIST option "ELEMENT TYPE" can only use named data types, but found unnamed data type int4 list. Use CREATE TYPE to create a named type first
+query error CREATE TYPE ... AS LIST option ELEMENT TYPE can only use named data types, but found unnamed data type \[s20 AS pg_catalog.int4\] list. Use CREATE TYPE to create a named type first
```
Comment (jkosh44) [inline test/sqllogictest/list.slt:1992]: These error messages with the global id are probably not what we want to be showing to the user. I'll open a follow up PR to fix them.

**Comment by def-** [comment]
Full SLT was green too: https://buildkite.com/materialize/sql-logic-tests/builds/5566#0188b3ba-3773-4371-b3a5-d3a54b120a37

**Pull request merged**